### ISSUE TITLE
how to disable shape compatibility check of TransformedDistribution

### ISSUE THREAD

**Issue by breadbread1984** [tensorflow/probability#528]
I implemented OpenAI's Glow whose input shape (event_shape) is different from output shape (sampled image shape). I create TransformedDistribution with following code.

```python
    glow = tfp.distributions.TransformedDistribution(
        distribution = tfp.distributions.Normal(loc = 0., scale = 1.),
        bijector = tfp.bijectors.Invert(Glow(input_shape = (32,32,1), levels = 2)),
        name = "transformed_dist",
        event_shape = (8,8,8) # shape of the random tensor generated by the base distribution
    );
```

When I sample from it, the transformed distribution try to reshape the sample into the same dimension of event shape and fails. How can I disable the reshape when I generate sample with different dimension from input event shape?

**Comment by brianwa84** [comment]
You can implement the _forward_event_shape{,_tensor} and
_inverse_event_shape{,_tensor} functions on your bijector.

The actual event size can't change except in special cases like
SoftmaxCentered, which goes from nd unconstrained space to a manifold in
(n+1)d space.

If you want to contribute an example or a notebook, we'd certainly welcome
glow!

**Comment by breadbread1984** [comment]
@brianwa84 here is my repo. Glow needs a transformed distribution pose no constraint on equality between dimensions of input and output dimension. For Glow, input random code has smaller dimension than output image's dimension.

**Comment by breadbread1984** [comment]
I found a way to bypass the problem. my glow is already workable.

**Comment by brianwa84** [comment]
The MatvecLU bijector should implement the invertible 1x1 conv for you, and you might want to try out RealNVP or MaskedAutoregressive as well. Glad you got things working.

**Comment by breadbread1984** [comment]
tfp's prebuilt bijectors are all dimension preserve. glow uses squeeze and split. therefore, the  tensor's dimension changes over the forward process. I just manually forward and inverse without using TransformedDistribution to avoid dimension compatibility check.

**Comment by brianwa84** [comment]
If I'm remembering correctly, glow just shifts the shape from height/width
into the depth axis, right? The total event size (product of dims) is
preserved. This is supported by tfb.Reshape.

**Comment by breadbread1984** [comment]
split operator will cut tensor to half. half for coming encoding layers, half for code.
so the code is generated at every split layer and the last layer. the event size changes.
